pkg/virtctl/console/console.go

```diff
@@ -135,11 +150,78 @@ func (c *Console) Run(flags *flag.FlagSet) int {
 
 	err = ws.WriteMessage(websocket.CloseMessage, websocket.FormatCloseMessage(websocket.CloseNormalClosure, ""))
 	if err != nil {
-		log.Fatalf("Error on close announcement: %s", err.Error())
+		return fmt.Errorf("Error on close announcement: %s", err.Error())
 	}
 	select {
 	case <-readStop:
 	case <-time.After(time.Second):
 	}
-	return 0
+	return nil
+}
+
+func requestFromConfig(config *rest.Config, vm string, namespace string, device string) (*http.Request, error) {
+
+	u, err := url.Parse(config.Host)
+	if err != nil {
+		return nil, err
+	}
+
+	switch u.Scheme {
+	case "https":
+		u.Scheme = "wss"
+	case "http":
+		u.Scheme = "ws"
+	default:
+		return nil, fmt.Errorf("Malformed URL %s", u.String())
```

Comment:
Is there a way to explain "we only support http and https" without getting too wordy? It might help users understand what's wrong with the URL.